blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpAggregatedIngestionHandler.java

```diff
@@ -61,65 +63,66 @@ public void handle(ChannelHandlerContext ctx, FullHttpRequest request) {
+                // no validation errors, process bundle
+                collection.add( PreaggregateConversions.buildMetricsCollection( payload ) );
+                ListenableFuture<List<Boolean>> futures = processor.apply( collection );
+                List<Boolean> persisteds = futures.get( timeout.getValue(), timeout.getUnit() );
+                for ( Boolean persisted : persisteds ) {
+                    if (!persisted) {
+                        DefaultHandler.sendErrorResponse(ctx, request, "Internal error persisting data", HttpResponseStatus.INTERNAL_SERVER_ERROR);
+                        return;
                     }
-                    DefaultHandler.sendResponse(ctx, request, null, HttpResponseStatus.OK);
-                } else {
-                    // has validation errors for the single metric, return BAD_REQUEST
-                    DefaultHandler.sendResponse(ctx,
-                            request,
-                            HttpMetricsIngestionHandler.getResponseBody(errors),
-                            HttpResponseStatus.BAD_REQUEST);
                 }
-
-            } catch (JsonParseException ex) {
-                log.debug(String.format("BAD JSON: %s", body));
-                log.error(ex.getMessage(), ex);
-                DefaultHandler.sendResponse(ctx, request, ex.getMessage(), HttpResponseStatus.BAD_REQUEST);
-            } catch (TimeoutException ex) {
-                DefaultHandler.sendResponse(ctx, request, "Timed out persisting metrics", HttpResponseStatus.ACCEPTED);
-            } catch (Exception ex) {
-                log.debug(String.format("JSON request payload: %s", body));
-                log.error("Error saving data", ex);
-                DefaultHandler.sendResponse(ctx, request, "Internal error saving data", HttpResponseStatus.INTERNAL_SERVER_ERROR);
-            } finally {
-                timerContext.stop();
+                DefaultHandler.sendResponse( ctx, request, null, HttpResponseStatus.OK );
+            } else {
+                // has validation errors for the single metric, return BAD_REQUEST
+                DefaultHandler.sendErrorResponse(ctx,
+                        request,
+                        validationErrors,
+                        HttpResponseStatus.BAD_REQUEST);
             }
+        } catch (JsonParseException ex) {
+            log.debug(String.format("BAD JSON: %s", body));
+            log.error(ex.getMessage(), ex);
+            DefaultHandler.sendErrorResponse(ctx, request, ex.getMessage(), HttpResponseStatus.BAD_REQUEST);
+        } catch (InvalidDataException ex) {
+            log.debug(String.format("Invalid request body: %s", body));
+            log.error(ex.getMessage(), ex);
+            DefaultHandler.sendErrorResponse(ctx, request, ex.getMessage(), HttpResponseStatus.BAD_REQUEST);
+        } catch (TimeoutException ex) {
+            DefaultHandler.sendErrorResponse(ctx, request, "Timed out persisting metrics", HttpResponseStatus.ACCEPTED);
```

Comment:
would we want to log an info or warn for this condition?

So basically we are optimistic here, right?  We are saying that the metrics haven't saved yet, but they will.